cli/azd/pkg/tools/azcli/keyvault.go

```diff
@@ -0,0 +1,133 @@
+package azcli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault"
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+	"github.com/azure/azure-dev/cli/azd/pkg/identity"
+)
+
+type AzCliKeyVault struct {
+	Id         string `json:"id"`
+	Name       string `json:"name"`
+	Location   string `json:"location"`
+	Properties struct {
+		EnableSoftDelete      bool `json:"enableSoftDelete"`
+		EnablePurgeProtection bool `json:"enablePurgeProtection"`
+	} `json:"properties"`
+}
+
+type AzCliKeyVaultSecret struct {
+	Id    string `json:"id"`
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+func (cli *azCli) GetKeyVault(ctx context.Context, subscriptionId string, resourceGroupName string, vaultName string) (*AzCliKeyVault, error) {
+	client, err := cli.createKeyVaultClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	vault, err := client.Get(ctx, resourceGroupName, vaultName, nil)
```

Comment:
This seems really odd that we need to pass `resourceGroupName` as the `keyvault` name itself is globally unique. Looking at the doc for [armkeyvault.KeyvaultClient.Get](
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault#SecretsClient.Get):

> Get - Gets the specified secret. NOTE: This API is intended for internal use in ARM deployments. Users should use the data-plane REST service for interaction with vault secrets. If the operation fails it returns an *azcore.ResponseError type. Generated from API version 2021-10-01 resourceGroupName - The name of the Resource Group to which the vault belongs. vaultName - The name of the vault. secretName - The name of the secret. options - SecretsClientGetOptions contains the optional parameters for the SecretsClient.Get method.

It seems to imply it's internal for ARM deployments. Is there a different API we should be using instead?

Reply:
I'll take a look to see if there is a different API we should be using.

Reply:
Looked deeper into this.  There isn't a different API to call.  When looking into `az cli` it performs a subscription wide resource list for all keyvaults and internally finds it by name.  It then performs the same API call with resource group from the list result.